src/apps/TxBuilder/hooks/useServices/index.tsx

```diff
@@ -0,0 +1,27 @@
+import Web3 from "web3";
+import InterfaceRepository from "./interfaceRepository";
+import { useSafe } from "../useSafe";
+
+import { InterfaceRepo } from "./interfaceRepository";
+
+export interface Services {
+  web3: any;
+  interfaceRepo: InterfaceRepo;
+}
+
+export default function useServices(): Services {
+  const safe = useSafe();
+
+  const web3 = new Web3(
```

Comment:
everytime useServices is called a new web3 object will be created, but it could be reused. Is it possible to optimize this?

Reply:
done :beer: